extensions/gitpod-shared/src/features.ts

```diff
@@ -474,12 +475,28 @@ export async function registerWorkspaceCommands(context: GitpodExtensionContext)
 			properties: { action: 'snapshot' }
 		});
 		try {
-			const snapshotId = await vscode.window.withProgress({
+			let snapshotId = '';
+			await vscode.window.withProgress({
```

Comment:
it looks like if a user press cancel still, either error or success notification with bogus link will be shown? Should not we guard?

Reply:
The code should be fine if we assume that `withProgress` bubbles up errors. But it's better to make that explicit, I agree.